src/main/java/de/tum/in/www1/artemis/web/rest/PlagiarismResource.java

```diff
@@ -1,46 +1,238 @@
+        }
+
+        if (comparison.getSubmissionA().getStudentLogin().equals(studentLogin)) {
+            plagiarismComparisonRepository.updatePlagiarismComparisonInstructorStatementA(comparison.getId(), instructorStatement);
+            comparison.setInstructorStatementA(instructorStatement); // needed for notifications
+        }
+        else if (comparison.getSubmissionB().getStudentLogin().equals(studentLogin)) {
+            plagiarismComparisonRepository.updatePlagiarismComparisonInstructorStatementB(comparison.getId(), instructorStatement);
+            comparison.setInstructorStatementB(instructorStatement); // needed for notifications
+        }
+        else {
+            throw new EntityNotFoundException("Student with id not found in plagiarism comparison");
+        }
+        singleUserNotificationService.notifyUserAboutNewPossiblePlagiarismCase(comparison, affectedUser);
+        return ResponseEntity.ok(statement);
+    }
+
+    /**
+     * Retrieves the plagiarismComparison specified by its Id. The submissions are anonymized for the student.
+     * StudentIds are replaced with "Your Submission" and "Other Submission" based on the requesting user.
+     *
+     * @param comparisonId the id of the PlagiarismComparison
+     * @return the PlagiarismComparison
+     * @throws AccessForbiddenException if the requesting user is not affected by the plagiarism case.
+     */
+    @GetMapping("plagiarism-comparisons/{comparisonId}")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<PlagiarismCaseDTO> getPlagiarismComparisonForStudent(@PathVariable("comparisonId") Long comparisonId) {
+        var comparison = plagiarismComparisonRepository.findByIdElseThrow(comparisonId);
+        User user = userRepository.getUser();
+
+        // check if current user is part of the comparison or not
+        if (!(comparison.getSubmissionA().getStudentLogin().equals(user.getLogin()) || comparison.getSubmissionB().getStudentLogin().equals(user.getLogin()))) {
+            throw new AccessForbiddenException("User tried updating plagiarism case they're not affected by.");
+        }
+
+        PlagiarismComparison anonymizedComparisonForStudentView = this.plagiarismService.anonymizeComparisonForStudentView(comparison, user.getLogin());
+        return ResponseEntity.ok(new PlagiarismCaseDTO(anonymizedComparisonForStudentView.getPlagiarismResult().getExercise(), Set.of(anonymizedComparisonForStudentView)));
+    }
+
+    /**
+     * Updates a student statement on a plagiarismComparison.
+     * I.e. one of the students that is accused of plagiarising updates/sets the respective/individual response/defence
+     *
+     * @param comparisonId of the comparison
+     * @param statement the students statement
+     * @return the student statement
+     */
+    @PutMapping("plagiarism-comparisons/{comparisonId}/student-statement")
```

Comment:
does not follow our REST URL best practices, please improve it:
- start with courses